plugins/gossip/index.js

```diff
@@ -191,6 +196,10 @@ module.exports = {
 
       }, 'string|object'),
 
+      handshakeTimeout: function(peer) {
```

Comment:
I don't think we should add another method just to have a logging event, (new methods should only be added very reluctantly!)

How do you feel about adding this as either an 'reason' option to disconnect, (which also causes a message to be logged), or to just log from gossip/schedule?

Reply:
I felt the handshake timeout was a 'real' event, ie. something that changed the state-machine of the client, so it seemed okay to make a new method. For example, what if the server did not want to disconnect on handshake timeout, but resend a hello message or otherwise restart the handshake without loosing the client connection.

A reason field on disconnect also seems fine (though I dont immediately understand the , 'string|object' syntax. grin. ), which ever you feel is best.

Reply:
oh yeah, some useful context: I think schedule shouldn't really be calling disconnect, that may be a hack because something else doesn't error properly. Also, if the timeout fails, it would be because of the tcp connection, and the thing that fails is either the tcp connection or the packets took too long in secret-handshake (authenticated encryption for that socket) it's hard to distinguish between a really slow tcp connection and a broken one, but in the p2p system it needs to be robust to breakages anyway, so just connect to another peer instead.